scrap.js

```diff
@@ -17,19 +19,82 @@ async function fetchLeaders(id) {
   return leaders
 }
 
-async function fetchOrgsWithLeaders() {
+async function searchGitHubOrgs(query) {
+  const token = process.env.GH_TOKEN
+  const res = await fetch(
+    `https://api.github.com/search/users?q=${query}%20type:org`,
+    {
+      headers: token ? { Authorization: `token ${token}` } : {}
+    }
+  )
+  const resp = await res.json()
+  const { items } = resp
+  return items
+}
+
+async function findOrganization({
+  name,
+  description,
+  mailing_list,
+  website_url,
+  irc_channel,
+  blog_url,
+  guide_to_working_url,
+}) {
+  if (overrides[name]) {
+    return overrides[name]
+  }
+
+  const githubPattern = /(?:https?:\/\/)?github\.com\/([a-zA-Z0-9]+)/i
+  const websites = [
+    mailing_list,
+    website_url,
+    irc_channel,
+    blog_url,
+    guide_to_working_url,
+  ]
+
+  const orgFromWebsites = websites
+    .map(website => (githubPattern.exec(website) || [])[1])
+    .filter(org => org)
```

Comment:
What's this filter?

Reply:
The map will return some `undefined` elements. This filter gets rid of them because `undefined` is falsey.

Reply:
Does the resulting array only contains 1 truthy element?
If so you can use `.find`.

Reply:
It may contain more than one element, but we only care about the first one, so you're right, `.find` is probably the better option!